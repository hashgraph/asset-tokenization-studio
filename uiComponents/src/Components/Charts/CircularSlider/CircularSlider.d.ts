import type { CircularProgressProps as ChakraCircularProgressProps } from "@chakra-ui/progress";
import React from "react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
export declare const circularSliderPartsList: ("label" | "track")[];
type Parts = typeof circularSliderPartsList;
export interface CircularSliderThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface CircularSliderProps extends Omit<ChakraCircularProgressProps, "children"> {
    label?: React.ReactElement | string;
    isLoading?: boolean;
}
export declare const CircularSlider: ({ label, size, isLoading, ...props }: CircularSliderProps) => JSX.Element;
export {};
