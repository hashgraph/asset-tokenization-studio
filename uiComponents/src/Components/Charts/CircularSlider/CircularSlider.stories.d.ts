declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./CircularSlider").CircularSliderProps>;
export default meta;
export declare const Small: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./CircularSlider").CircularSliderProps>;
export declare const Medium: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./CircularSlider").CircularSliderProps>;
export declare const WithLabel: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./CircularSlider").CircularSliderProps>;
