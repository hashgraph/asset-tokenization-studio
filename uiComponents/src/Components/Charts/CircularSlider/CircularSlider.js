var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { SkeletonCircle as ChakraSkeletonCircle } from "@chakra-ui/react";
import React from "react";
import { CircularProgress as ChakraCircularProgress, CircularProgressLabel as ChakraCircularProgressLabel, } from "@chakra-ui/progress";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
export var circularSliderPartsList = [
    "label",
    "track",
];
export var CircularSlider = function (_a) {
    var label = _a.label, size = _a.size, isLoading = _a.isLoading, props = __rest(_a, ["label", "size", "isLoading"]);
    var styles = useChakraMultiStyleConfig("CircularSlider", { size: size });
    if (isLoading) {
        return React.createElement(ChakraSkeletonCircle, { sx: styles.track });
    }
    return (React.createElement(ChakraCircularProgress, __assign({ sx: styles.track }, props), label !== undefined && (React.createElement(ChakraCircularProgressLabel, { sx: styles.label }, label))));
};
