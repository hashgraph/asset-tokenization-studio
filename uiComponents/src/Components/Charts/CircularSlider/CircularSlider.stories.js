var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { CircularSlider } from "./CircularSlider";
var meta = {
    title: "Design System/Charts/CircularSlider",
    component: CircularSlider,
    args: {
        size: "sm",
        color: "blue.500",
        value: 40,
        isLoading: false
    },
    argTypes: {
        label: {
            control: { type: "text" }
        },
        value: {
            control: { type: "number" }
        },
        size: {
            options: ["sm", "md"],
            control: { type: "radio" }
        },
        color: {
            control: { type: "color" }
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A8803"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return (React.createElement(CircularSlider, __assign({}, args))); };
export var Small = Template.bind({});
Small.args = {
    size: "sm"
};
export var Medium = Template.bind({});
Medium.args = {
    size: "md"
};
export var WithLabel = Template.bind({});
WithLabel.args = {
    label: "40%"
};
