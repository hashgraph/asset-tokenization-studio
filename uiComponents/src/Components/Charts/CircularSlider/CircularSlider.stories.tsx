import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { CircularSlider } from "./CircularSlider";

const meta = {
  title: "Design System/Charts/CircularSlider",
  component: CircularSlider,
  args: {
    size: "sm",
    color: "blue.500",
    value: 40,
    isLoading: false,
  },
  argTypes: {
    label: {
      control: { type: "text" },
    },
    value: {
      control: { type: "number" },
    },
    size: {
      options: ["sm", "md"],
      control: { type: "radio" },
    },
    color: {
      control: { type: "color" },
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A8803",
    },
    docs: {},
  },
} as Meta<typeof CircularSlider>;
export default meta;

const Template: StoryFn<typeof CircularSlider> = (args) => (
  <CircularSlider {...args} />
);

export const Small = Template.bind({});
Small.args = {
  size: "sm",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "md",
};

export const WithLabel = Template.bind({});
WithLabel.args = {
  label: "40%",
};
