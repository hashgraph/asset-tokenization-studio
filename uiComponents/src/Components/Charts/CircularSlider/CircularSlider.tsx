import type { CircularProgressProps as ChakraCircularProgressProps } from "@chakra-ui/progress";
import { SkeletonCircle as ChakraSkeletonCircle } from "@chakra-ui/react";
import React from "react";

import {
  CircularProgress as ChakraCircularProgress,
  CircularProgressLabel as ChakraCircularProgressLabel,
} from "@chakra-ui/progress";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";

export const circularSliderPartsList: ("label" | "track")[] = [
  "label",
  "track",
];

type Parts = typeof circularSliderPartsList;

export interface CircularSliderThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export interface CircularSliderProps
  extends Omit<ChakraCircularProgressProps, "children"> {
  label?: React.ReactElement | string;
  isLoading?: boolean;
}

export const CircularSlider = ({
  label,
  size,
  isLoading,
  ...props
}: CircularSliderProps) => {
  const styles = useChakraMultiStyleConfig("CircularSlider", { size });

  if (isLoading) {
    return <ChakraSkeletonCircle sx={styles.track} />;
  }

  return (
    <ChakraCircularProgress sx={styles.track} {...props}>
      {label !== undefined && (
        <ChakraCircularProgressLabel sx={styles.label}>
          {label}
        </ChakraCircularProgressLabel>
      )}
    </ChakraCircularProgress>
  );
};
