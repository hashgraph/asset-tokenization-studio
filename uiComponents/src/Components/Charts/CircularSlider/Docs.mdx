import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as CircularSliderStories from "./CircularSlider.stories";

<Meta of={CircularSliderStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=3591%3A55878&mode=design&t=k4jSCm0Q182mRyVn-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `CircularSlider.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    CircularSlider: ConfigCircularSlider,
  },
};
```

```tsx dark
const ConfigCircularSlider: CircularSliderThemeConfiguration = {
  parts: circularSliderPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"label" | "track"` (`circularSliderPartsList`)

## Props

<ArgTypes of={CircularSliderStories} />

## Stories

<Stories title="" />
