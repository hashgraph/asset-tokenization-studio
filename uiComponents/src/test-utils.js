import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
