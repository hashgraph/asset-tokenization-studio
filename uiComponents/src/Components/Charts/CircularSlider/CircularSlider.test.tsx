import { render } from "@/test-utils";
import React from "react";
import type { CircularSliderProps } from "./CircularSlider";
import { CircularSlider } from "./CircularSlider";

describe(`<CircularSlider />`, () => {
  const textLabel = "radio item";
  const labelProp = { label: textLabel };
  const valueProp = { value: 40 };
  const sizeSmallProp = { size: "sm" };
  const sizeMediumProp = { size: "md" };
  const colorProp = { color: "red" };

  const factoryComponent = (props?: CircularSliderProps) =>
    render(<CircularSlider {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly when a label is passed over it ", () => {
    const component = factoryComponent(labelProp);

    expect(component.getByText(textLabel)).toBeVisible();

    expect(component.asFragment()).toMatchSnapshot("Using label");
  });

  test("shows correctly when a value is passed over it ", () => {
    const component = factoryComponent(valueProp);

    expect(component.asFragment()).toMatchSnapshot("Using value");
  });

  test("shows correctly when size is small", () => {
    const component = factoryComponent(sizeSmallProp);

    expect(component.asFragment()).toMatchSnapshot("Using small size");
  });

  test("shows correctly when size is medium", () => {
    const component = factoryComponent(sizeMediumProp);

    expect(component.asFragment()).toMatchSnapshot("Using medium size");
  });

  test("shows correctly when a color is passed over it ", () => {
    const component = factoryComponent(colorProp);

    expect(component.asFragment()).toMatchSnapshot("Using color");
  });

  test("shows correctly when isLoading is true", () => {
    const component = factoryComponent({ isLoading: true });

    expect(component.asFragment()).toMatchSnapshot("Using isLoading");
  });
});
