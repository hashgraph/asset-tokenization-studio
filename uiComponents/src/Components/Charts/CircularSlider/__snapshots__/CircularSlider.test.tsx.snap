// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularSlider /> renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="chakra-progress css-eb5xc8"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-1thbxp8"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="0 264"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when a color is passed over it : Using color 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="chakra-progress css-eb5xc8"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-1prp7l4"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="0 264"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when a label is passed over it : Using label 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="chakra-progress css-eb5xc8"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-1thbxp8"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="0 264"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
    <div
      class="css-1uer2ql"
    >
      radio item
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when a value is passed over it : Using value 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="40"
    class="chakra-progress css-eb5xc8"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-hvagwd"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="105.60000000000001 158.39999999999998"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when isLoading is true: Using isLoading 1`] = `
<DocumentFragment>
  <div
    class="chakra-skeleton css-1xm5d29"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when size is medium: Using medium size 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="chakra-progress css-ly9idg"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-1thbxp8"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="0 264"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CircularSlider /> shows correctly when size is small: Using small size 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="chakra-progress css-eb5xc8"
    role="progressbar"
  >
    <svg
      class="css-kn46u7"
      viewBox="0 0 100 100"
    >
      <circle
        class="chakra-progress__track css-ol3i12"
        cx="50"
        cy="50"
        r="42"
        stroke-width="10px"
      />
      <circle
        class="chakra-progress__indicator css-1thbxp8"
        cx="50"
        cy="50"
        r="42"
        stroke-dasharray="0 264"
        stroke-dashoffset="66"
        stroke-width="10px"
      />
    </svg>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
